frontend/public/components/resource-dropdown.tsx

```diff
@@ -198,58 +245,217 @@ const ResourceListDropdown_: React.SFC<ResourceListDropdownProps> = (props) => {
+    const focusedItem = focusedItemIndex !== null ? selectOptions[focusedItemIndex] : null;
+
+    // eslint-disable-next-line default-case
+    switch (event.key) {
+      case 'Enter':
+        if (
+          isOpen &&
+          focusedItem &&
+          focusedItem.value !== NO_RESULTS &&
+          !focusedItem.isAriaDisabled
+        ) {
+          onSelect(focusedItem.value);
+        }
+
+        if (!isOpen) {
+          setIsOpen(true);
+        }
+
+        break;
+      case 'ArrowUp':
+      case 'ArrowDown':
+        event.preventDefault();
+        handleMenuArrowKeys(event.key);
+        break;
+    }
+  };
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+    textInputRef?.current?.focus();
   };
 
-  const onToggle = (_event, newOpenState: boolean) => {
-    setOpen(newOpenState);
+  const onClearButtonClick = () => {
+    setInputValue('');
+    resetActiveAndFocusedItem();
+    textInputRef?.current?.focus();
   };
 
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle ref={toggleRef} variant="typeahead" onClick={onToggleClick} isExpanded={isOpen}>
+      <TextInputGroup isPlain>
+        <TextInputGroupMain
+          value={inputValue}
+          onClick={onInputClick}
+          onChange={onTextInputChange}
+          onKeyDown={onInputKeyDown}
+          id="typeahead-select-input"
+          autoComplete="off"
+          innerRef={textInputRef}
+          placeholder={placeholder}
+          {...(activeItemId && { 'aria-activedescendant': activeItemId })}
+          role="combobox"
+          isExpanded={isOpen}
+          aria-controls="select-typeahead-listbox"
+        />
+
+        <TextInputGroupUtilities {...(!inputValue ? { style: { display: 'none' } } : {})}>
+          <Button variant="plain" onClick={onClearButtonClick} aria-label="Clear input value">
+            <TimesIcon aria-hidden />
+          </Button>
+        </TextInputGroupUtilities>
+      </TextInputGroup>
+    </MenuToggle>
+  );
+
   return (
-    <SelectDeprecated
-      variant={SelectVariantDeprecated.checkbox}
-      onToggle={onToggle}
-      onSelect={handleSelected}
-      selections={selectedOptions}
-      isOpen={isOpen}
-      placeholderText={t('public~Resources')}
-      inlineFilterPlaceholderText={t('public~Select Resource')}
-      onFilter={onCustomFilter}
-      hasInlineFilter
-      customBadgeText={selected.length}
-      className={classNames('co-type-selector', className)}
-      maxHeight="60vh"
-      isGrouped
-    >
-      {renderedOptions()}
-    </SelectDeprecated>
+    <div className={className}>
+      <Select
+        toggle={toggle}
+        onSelect={(_event, selection) => onSelect(selection as string)}
+        selected={selected}
+        isOpen={isOpen}
+        onOpenChange={(open) => {
+          !open && closeMenu();
+        }}
+        // maxMenuHeight="60vh" A bug in the current PatternFly version prevents usage
```

Comment:
@sg00dwin, have you run in to this yet?  I'm assuming it [has been fixed](https://github.com/patternfly/patternfly-react/issues?q=is%3Aissue+is%3Aclosed+maxMenuHeight), we're just not using a version that includes the fix.

Reply:
@rhamilto I hadn't hit that yet. But based on there docs it should work.

Reply:
Added CSS to address so we retain existing behavior.